src/aiogram_dialog/api/protocols/dialog.py

```diff
@@ -1,11 +1,13 @@
 from abc import abstractmethod
-from typing import Any, Dict, List, Optional, Protocol, runtime_checkable, Type
+from typing import Any, Dict, List, Optional, Protocol, Type, Union, runtime_checkable
 
 from aiogram.fsm.state import State, StatesGroup
 
-from aiogram_dialog.api.entities import (
-    Data, LaunchMode, NewMessage,
-)
+from aiogram_dialog.api.entities import Data, LaunchMode, NewMessage
+from ..internal import Widget
+
+from ... import ChatEvent
```

Comment:
1. Please, avoid `...` imports. I would allow relative import within widgets package, but not outside of it.
2. Do not import from library package itself, import from subpackages, otherwise it will lead to cycle imports